src/main/scala/org/scalacheck/Gen.scala

```diff
@@ -656,80 +666,84 @@ object Gen extends GenArities with GenVersionSpecific {
   ): Gen[C] =
-    sized(s => choose(1, s max 1).flatMap(buildableOfN[C,T](_,g))) suchThat(c => evt(c).size > 0)
+    buildableOf(g)(evb, evt).suchThat(c => evt(c).size > 0)
 
   /** A convenience method for calling `buildableOfN[C[T],T](n,g)`. */
   def containerOfN[C[_],T](n: Int, g: Gen[T])(implicit
     evb: Buildable[T,C[T]], evt: C[T] => Traversable[T]
-  ): Gen[C[T]] = buildableOfN[C[T],T](n,g)
+  ): Gen[C[T]] = buildableOfN[C[T], T](n, g)(evb, evt)
 
   /** A convenience method for calling `buildableOf[C[T],T](g)`. */
   def containerOf[C[_],T](g: Gen[T])(implicit
     evb: Buildable[T,C[T]], evt: C[T] => Traversable[T]
-  ): Gen[C[T]] = buildableOf[C[T],T](g)
+  ): Gen[C[T]] = buildableOf[C[T], T](g)(evb, evt)
 
   /** A convenience method for calling `nonEmptyBuildableOf[C[T],T](g)`. */
   def nonEmptyContainerOf[C[_],T](g: Gen[T])(implicit
     evb: Buildable[T,C[T]], evt: C[T] => Traversable[T]
-  ): Gen[C[T]] = nonEmptyBuildableOf[C[T],T](g)
+  ): Gen[C[T]] = nonEmptyBuildableOf[C[T], T](g)(evb, evt)
 
   /** Generates a list of random length. The maximum length depends on the
    *  size parameter. This method is equal to calling
    *  `containerOf[List,T](g)`. */
-  def listOf[T](g: => Gen[T]) = buildableOf[List[T],T](g)
+  def listOf[T](g: => Gen[T]) = buildableOf[List[T], T](g)
 
   /** Generates a non-empty list of random length. The maximum length depends
    *  on the size parameter. This method is equal to calling
    *  `nonEmptyContainerOf[List,T](g)`. */
-  def nonEmptyListOf[T](g: => Gen[T]) = nonEmptyBuildableOf[List[T],T](g)
+  def nonEmptyListOf[T](g: => Gen[T]) = nonEmptyBuildableOf[List[T], T](g)
 
   /** Generates a list with at most the given number of elements. This method
    *  is equal to calling `containerOfN[List,T](n,g)`. */
-  def listOfN[T](n: Int, g: Gen[T]) = buildableOfN[List[T],T](n,g)
+  def listOfN[T](n: Int, g: Gen[T]) = buildableOfN[List[T], T](n, g)
 
   /** Generates a map of random length. The maximum length depends on the
    *  size parameter. This method is equal to calling
    *  <code>containerOf[Map,(T,U)](g)</code>. */
-  def mapOf[T,U](g: => Gen[(T,U)]) = buildableOf[Map[T,U],(T,U)](g)
+  def mapOf[T, U](g: => Gen[(T, U)]) = buildableOf[Map[T, U], (T, U)](g)
 
   /** Generates a non-empty map of random length. The maximum length depends
    *  on the size parameter. This method is equal to calling
    *  <code>nonEmptyContainerOf[Map,(T,U)](g)</code>. */
-  def nonEmptyMap[T,U](g: => Gen[(T,U)]) = nonEmptyBuildableOf[Map[T,U],(T,U)](g)
+  def nonEmptyMap[T,U](g: => Gen[(T,U)]) = nonEmptyBuildableOf[Map[T, U],(T, U)](g)
 
   /** Generates a map with at most the given number of elements. This method
    *  is equal to calling <code>containerOfN[Map,(T,U)](n,g)</code>. */
-  def mapOfN[T,U](n: Int, g: Gen[(T,U)]) = buildableOfN[Map[T,U],(T,U)](n,g)
+  def mapOfN[T,U](n: Int, g: Gen[(T, U)]) = buildableOfN[Map[T, U],(T, U)](n, g)
 
   /** Generates an infinite stream. */
```

Comment:
isn't this a lie? Doesn't it stop at the first failure?

Again, here is a place for a retry strategy: we may try to retry some number of times.

Or: we could just call `unfold(p, r.seed)` on the `None` branch: presumably we will hit more items... It isn't clear to me why a single filter should kill the infinite stream.

Reply:
Yeah, the comment should have said "Generates a potentially infinite stream"

Retrying seems fine to me. I didn't change the existing behavior.

Reply:
No, `buildableOfN` does retrying and only returns `Some(_)` if it ran the underlying generator N times successfully.

There is a _different_ issue for sets/maps, which is that N generator runs doesn't mean you got N distinct items/keys. That is worth fixing, but this change preserves the previous logic.

Reply:
The harder problem for `buildableOfN` is that if you ask for `buildableOfN[Set[Byte], Byte](256, Gen.choose(Byte.MinValue, Byte.MaxValue))` the naive implementation will either mostly fail, or run for a very long time.

Reply:
(Sorry, Github's layout confused me and I left some comments here that didn't make sense, which I've since deleted.)